### PR TITLE
Sddmm implementation & tests

### PR DESCRIPTION
**Pull request by andrewpareles**

### CODE REVIEW

**Comment by zzy82518996** [comment]
Looks good to me !

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -6624,3 +6624,10 @@
 #   use_c10_dispatcher: full
 #   variants: function
 #   dispatch:
+
+- func: sddmm(Tensor self, Tensor mat1, Tensor mat2) -> Tensor
+  use_c10_dispatcher: full
+  variants: function, method
+  dispatch:
+    # SparseCPU: sddmm_cpu
+    SparseHammerBlade: sddmm_hb
```
Comment (yodada) [inline aten/src/ATen/native/native_functions.yaml]: looks like you are missing a newline at the end of native_functions.yaml

**Review by sampsyo — APPROVED** [review]
Yay!! Looks fantastic to me.

**Review by yodada — APPROVED** [review]
Looks good! The flake8 error is not in the change set so ...

**Pull request merged**